src/main/java/sg/ncl/MainController.java

```diff
@@ -1532,6 +1535,91 @@ public String rejectJoinRequest(@PathVariable Integer teamId, @PathVariable Inte
+                                                                                " team leader or creator of this image can delete the image.");
+                        return "redirect:/teams";
+                    case TEAM_NOT_FOUND_EXCEPTION:
+                        log.warn("Error in deleting image {} from team '{}' : team not found", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "Team '" + teamId + "' is not found");
+                        return "redirect:/teams";
+                    case IMAGE_NOT_FOUND_EXCEPTION:
+                        log.warn("Error in deleting image {} from team '{}' : image does not exist or not found in teams' list of images", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "The image '" + imageName + "' either does not exist or not found in your teams' list of images");
+                        return "redirect:/teams";
+                    case DETERLAB_OPERATION_FAILED_EXCEPTION:
+                        log.warn("Error in deleting image '{}' from team '{}' : operation failed on DeterLab", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    case ADAPTER_CONNECTION_EXCEPTION:
+                        log.warn("Error in deleting  image '{}' from team '{}' : adapter connection error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    case ADAPTER_INTERNAL_ERROR_EXCEPTION:
+                        log.warn("Error in deleting image '{}' from team '{}' : adapter internal error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    default:
+                        log.warn("Error in deleting image '{}' from team '{}' : {}", imageName, teamId,  exceptionState);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                }
+            } else {
+                String sioMessage = new JSONObject(responseBody).getString("msg");
+
+                switch (sioMessage) {
+                    case "image still in use":
+                        log.warn("Error in deleting image '{}' from team '{}' : {}", imageName, teamId, sioMessage);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "Image " + "\'" + imageName + "\'" + " is still in use or busy!");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Image " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)